runtime/support.py

```diff
@@ -405,8 +417,12 @@ def setup(import_roots, zip_safe, extract_dir=None):
     # Initialize import path
     _initialize_import_path(import_roots, import_prefix)
 
-    # Add hook for package metadata
-    _setup_pkg_resources('pkg_resources')
-    _setup_pkg_resources('pip._vendor.pkg_resources')
+    # If we are using unpar mode, pkg_resources should be able to find files
+    # like it normally does, and we don't need to do this import here (which is
+    # slow and unnecessary if the payload doesn't use pkg_resources).
+    if not extract_dir:
```

Comment:
Hmm, I wonder actually if this should say `if zip_safe`, since we could probably also skip this step when extracting to a tmpdir, right? Needs testing, I suppose

Reply:
for our usage it doesn't really matter which one you use

Reply:
yeah, I guess I was more wondering if there might be some corner cases where we still do use tmpdirs, like the case we once had for PCLI running as non-root users... but I guess if we do have instances like that, we should just fix them to use `PAR_EXTRACT_ONLY` instead of worrying about it here.